### DISCUSSION TITLE
Why isn't react and react-dom peer dependencies ?

### DISCUSSION THREAD

**Discussion by stevebeauge** [mantinedev/mantine#5940] [Q&A]
Hello,

I'm new to mantine. I'm evaluating the opportunity to migrate our apps from other 3rd party ui libs to mantine, which looks like promising.

However, I'm stuck to react 17 because of some legacy apps and libraries. Mantine declares react and react-dom as direct dependencies, instead of peer dependencies, which result in invalid dependency tree.

Is there any reason to have such high requirements? Is there any workaround to allow using mantine in a react 17 app?

PS: to give a bit of context, we are actually developing addons for the Microsoft 365 platform (SPFX), which uses react 17, which is unlikely to change before months or years.

**Comment by rtivital** [comment]
I do not understand the question. `react` and `react-dom` are listed as `peerDependencies` in all packages – https://github.com/mantinedev/mantine/blob/master/packages/%40mantine/core/package.json#L45-L48

**Comment by stevebeauge** [comment]
You are right. I focused on the root [workspace's package.json](https://github.com/mantinedev/mantine/blob/master/package.json), which is not the one used when consuming the library. 
Sorry for that.

However, I'm still looking for a way to use the lib using older versions of react. I guess there are some apis in react 18 that make mandatory this version.

**Comment by stevebeauge** [comment]
I tried another approach, which consists in embedding my own version of react.
This is out of scope of this library, so I close the question.

Thanks for you feedback anyways
